doc/src/modules/physics/mechanics/examples/four_bar_linkage_example.rst

```diff
@@ -48,30 +56,28 @@ Next, we also define the first three joints. ::
    >>> joint3 = PinJoint('J3', link3, link4, coordinates=q3, speeds=u3,
    ...                   parent_point=l3 / 2 * link3.x,
    ...                   child_point=-l4 / 2 * link4.x, joint_axis=link3.z)
+   >>> system.add_joints(joint1, joint2, joint3)
 
 Now we can formulate the holonomic constraint that will close the kinematic
 loop. ::
 
-   >>> loop = link4.masscenter.pos_from(link1.masscenter) + l1 / 2 * link1.x + l4 / 2 * link4.x
-   >>> fh = Matrix([loop.dot(link1.x), loop.dot(link1.y)])
+   >>> loop = (link4.masscenter.pos_from(link1.masscenter) +
+   ...         l1 / 2 * link1.x +l4 / 2 * link4.x)
+   >>> system.add_holonomic_constraints(loop.dot(link1.x), loop.dot(link1.y))
 
-In order to generate the equations of motions, we will use the ``JointsMethod``
-as our fronted. Before setting up the :class:`~.KanesMethod` as its backend we
-need to calculate the velocity constraints. ::
+Before generating the equations of motion we need to specify which generalized
+coordinates and speeds are independent and which are dependent. After which we
+can run :meth:`~.System.validate_system` to do some basic consistency checks. ::
 
-   >>> method = JointsMethod(link1, joint1, joint2, joint3)
-   >>> t = dynamicsymbols._t
-   >>> qdots = solve(method.kdes, [q1.diff(t), q2.diff(t), q3.diff(t)])
-   >>> fhd = fh.diff(t).subs(qdots)
+   >>> system.q_ind = [q1]
+   >>> system.u_ind = [u1]
+   >>> system.q_dep = [q2, q3]
+   >>> system.u_dep = [u2, u3]
+   >>> system.validate_system()
 
-Now we can setup the :class:`~.KanesMethod` as the backend and compute the
-equations of motion. ::
+As we have the entire system ready, we can now form the equations of motion
+using :class:`~.KanesMethod` as the backend. ::
 
-   >>> method._method = KanesMethod(
-   ...     method.frame, q_ind=[q1], u_ind=[u1], q_dependent=[q2, q3],
-   ...     u_dependent=[u2, u3], kd_eqs=method.kdes,
-   ...     configuration_constraints=fh, velocity_constraints=fhd,
-   ...     forcelist=method.loads, bodies=method.bodies)
-   >>> simplify(method.method._form_eoms())
+   >>> simplify(system.form_eoms())
```

Comment:
This seems to return only the dynamical differential equations, right? I would expect "equations of motion" to giv me the entire equations of motion (including kinematical too).

Also, can System use any equation of motion derivation method? Would this return the Lagrange multiplier form for lagrange's method, for example?

Reply:
Maybe this shouldn't return anything and you simply access the updated properties on the class. Then the name wouldn't slightly mismatch what it returns.